src/station_cmd.cpp

```diff
@@ -1531,12 +1529,8 @@ CommandCost CmdBuildRailStation(DoCommandFlags flags, TileIndex tile_org, RailTy
 		}
 
 		Track track = AxisToTrack(axis);
-
-		auto it = stl.begin();
-
 		Company *c = Company::Get(st->owner);
-		TileIndex tile_track = tile_org;
-		for (uint i = 0; i != numtracks; ++i) {
+		for (auto [i, it, tile_track] = std::make_tuple(0, stl.begin(), tile_org); i != numtracks; ++i, tile_track += track_delta) {
 			TileIndex tile = tile_track;
 			for (uint j = 0; j != plat_len; ++j) {
```

Comment:
```suggestion
			for (uint j = 0; j != plat_len; ++j, tile += tile_delta) {
```
and line 1586 can be removed.

(All this code is a pain to follow)